modules/node-shared/src/main/scala/org/tessellation/node/shared/domain/seedlist/HistoricalSnapshotOrdinalEntries.scala

```diff
@@ -0,0 +1,79 @@
+package org.tessellation.node.shared.domain.seedlist
+
+import cats.data.NonEmptyList
+import cats.syntax.all._
+
+import scala.util.control.NoStackTrace
+
+import org.tessellation.schema.SnapshotOrdinal
+
+import derevo.cats.{eqv, show}
+import derevo.derive
+import eu.timepit.refined.auto._
+import eu.timepit.refined.cats._
+import eu.timepit.refined.types.numeric.NonNegLong
+import fs2.data.csv.{CellDecoder, DecoderError}
+
+@derive(eqv)
+case class HistoricalSnapshotOrdinalEntries(values: Option[NonEmptyList[HistoricalSnapshotOrdinalEntry]])
```

Comment:
What's the advantage of having an optional non-empty list vs a regular list that could be empty?
Also, there's this:
```
case class SeedlistEntry(
  peerId: PeerId,
  ipAddress: Option[IpAddress],
  alias: Option[Alias],
  bias: Option[TrustValueRefined],
  ordinalTimeline: Option[HistoricalSnapshotOrdinalEntries]
)
```
In other words,
```
   ordinalTimeline: Option[HistoricalSnapshotOrdinalEntries]
=> ordinalTimeline: Option[Option[NonEmptyList[HistoricalSnapshotOrdinalEntry]]]
```
What does that mean?

Reply:
@sdavidp sometimes we want to distinguish cases when list exists but is empty from list not existing. It is similar as having `Int` set to `0` vs having `Option[PosInt]`.

Having an `Option` forces you to check whether you have some elements or not but with regular list compile won't force you to do that and you end up with empty list (which is incorrect data according to the logic written by Chuong). If you look at my previous comments you will see that Chuong did something like this:

`.toList.some.filter(_.nonEmpty)`

Which sounds to me like `toNel`. That's why I suggested using `NonEmptyList` (so we are sure that is has elements just by reading it) in the same manner that we use `PosInt` (so we know just by reading that is is positive) instead of having `Option[Int]` and checking if it is greater than `0`.